examples/06-mnist.rs

```diff
@@ -98,8 +98,8 @@ fn main() {
     let mut rng = StdRng::seed_from_u64(0);
 
     // initialize model and optimizer
-    let mut model: Mlp = dev.build_module();
-    let mut opt: Adam<Mlp> = Default::default();
```

Comment:
Users probably shouldn't specify `Adam<Mlp>` anymore, since that will change based on device. We'll probably need to not allow Default anymore and pass in what you want to optimize to the optimizer constructor. This should throw a helpful error if you pass in the wrong thing.